### PR TITLE
Handle eventcode for kafka and service registry

### PR DESCRIPTION
**Pull request by ajaypratap003**

### CODE REVIEW

File: src/app/pages/ServiceRegistry/ServiceRegistryPage.tsx [hunk 1]
```diff
@@ -19,7 +20,7 @@ export const ServiceRegistryPage: React.FunctionComponent = () => {
 export const ServiceRegistryPageConnected: React.FC = () => {
   const config = useConfig();
   const { getQuota } = useQuota(ProductType.srs);
-  const { preCreateInstance, shouldOpenCreateModal } = useModalControl();
+  const { preCreateInstance, shouldOpenCreateModal } = useModalControl(termsConfig['service-registry'].EventCode);
```
Comment (pmuir) [inline src/app/pages/ServiceRegistry/ServiceRegistryPage.tsx:23]: Can you add a `type` for the `termsConfig`?

File: src/app/hooks/modal.ts [hunk 2]
```diff
@@ -3,8 +3,8 @@ import { parse as parseQueryString, stringifyUrl } from 'query-string';
 import { useAsyncTermsReview } from '@app/services/termsReview';
 import { getTermsAppURL } from '@app/utils/termsApp';
 
-export const useModalControl = () => {
-    const loadTermsReview = useAsyncTermsReview();
+export const useModalControl = (eventCode: string) => {
```
Comment (pmuir) [inline src/app/hooks/modal.ts:6]: do the same for the siteCode - though you can pass a single type

Comment (ajaypratap003) [reply src/app/hooks/modal.ts]: done

File: src/app/pages/ServiceRegistry/ServiceRegistryPage.tsx (hunk 1 above)
Comment (ajaypratap003) [reply src/app/pages/ServiceRegistry/ServiceRegistryPage.tsx]: done

**Review by pmuir — APPROVED** [review]

**Pull request merged**